apps/web/src/utils/hooks/debounced.ts

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useState } from 'react';
+
+export const useDebounce = (value: unknown, delay: number) => {
+	const [debounced, setDebounced] = useState(value);
+
+	useEffect(() => {
```

Comment:
not yet debounced, this code will actually run `setDebounced` all the time with delay. Run`clearTimeout` on effect cleanup not yet enough, will need to run in `clearTimeout` every single dependency changes also.

What I thought, just for idea.. could be in-accurate:
```typescript 
const handler = useRef();

useEffect(() => {
  clearTimeout(handler.current); // cancel any timeout if it is pending/existed
  handler.current = setTimeout(() => {...}, delay);
  
  return () => clearTimeout(handler.current);
}, [value, delay]);
```

Reply:
Yes Sir! I realized this last night and I will refactor this hook to work effeciently